recipe/patch_yaml/hdf5.yaml

```diff
@@ -2,8 +2,8 @@
 # hdf5 1.14.2 and 1.14.3 have the same ABI
 # https://gamma.hdfgroup.org/ftp/pub/outgoing/hdf5/prelim_reports/hdf5-1.14.2-vs-hdf5-1.14.3-1-interface_compatibility_report.html
 if:
-  has_depends: hdf5 >=1.14.2,<1.14.3.0a0
-  timestamp_lt: 1701868616000
+  has_depends: hdf5 >=1.14.2,<1.14.3.0a0?( *)
+  timestamp_lt: 1702066879000
```

Comment:
Could you please share a bit about the timestamp change? It appears the new timestamp is more recent than the previous one

Reply:
Wonder if we should just rebuild the `hdf5` packages to include a better `run_exports` line

While in theory dropping the timestamp condition would mean anything would be patched. Really it would mean we patch the repodata based on whenever this feedstock last built a package with a new patch. Though when that happens is kind of arbitrary

Reply:
I think this is really out of scope for now.

We should really just answer this question in my mind:
https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/3368

For now this PR looks like a step in the right direction. the timestamp still exists so itisn't so wide reaching, but there is still a risk of systems being built with 1.14.2 instead of 1.14.3 that would carry the 'x.x.x' limitations